src/main/scala/shell/xilinx/VCU118NewShell.scala

```diff
@@ -460,27 +463,38 @@ abstract class VCU118ShellBasicOverlays()(implicit p: Parameters) extends UltraS
 }
 
 case object VCU118ShellPMOD extends Field[String]("JTAG")
+case object VCU118ShellPMOD2 extends Field[String]("JTAG")
 
 class WithVCU118ShellPMOD(device: String) extends Config((site, here, up) => {
   case VCU118ShellPMOD => device
 })
 
+// Change JTAG pinouts to VCU118 J53
+// Due to the level shifter is from 1.2V to 3.3V, the frequency of JTAG should be slow down to 1Mhz
+class WithVCU118ShellPMOD2(device: String) extends Config((site, here, up) => {
+  case VCU118ShellPMOD2 => device
+})
+
 class WithVCU118ShellPMODJTAG extends WithVCU118ShellPMOD("JTAG")
 class WithVCU118ShellPMODSDIO extends WithVCU118ShellPMOD("SDIO")
 
+// Reassign JTAG pinouts location to PMOD J53
+class WithVCU118ShellPMOD2JTAG extends WithVCU118ShellPMOD2("PMODJ53_JTAG")
+
 class VCU118Shell()(implicit p: Parameters) extends VCU118ShellBasicOverlays
 {
   val pmod_is_sdio  = p(VCU118ShellPMOD) == "SDIO"
-  val jtag_location = Some(if (pmod_is_sdio) "FMC_J2" else "PMOD_J52")
+  val pmod_j53_is_jtag = p(VCU118ShellPMOD2) == "PMODJ53_JTAG"
+  val jtag_location = Some(if (pmod_is_sdio) (if (pmod_j53_is_jtag) "PMOD_J53" else "FMC_J2") else "PMOD_J52")
 
   // Order matters; ddr depends on sys_clock
   val uart      = Overlay(UARTOverlayKey, new UARTVCU118ShellPlacer(this, UARTShellInput()))
   val sdio      = if (pmod_is_sdio) Some(Overlay(SPIOverlayKey, new SDIOVCU118ShellPlacer(this, SPIShellInput()))) else None
   val jtag      = Overlay(JTAGDebugOverlayKey, new JTAGDebugVCU118ShellPlacer(this, JTAGDebugShellInput(location = jtag_location)))
   val cjtag     = Overlay(cJTAGDebugOverlayKey, new cJTAGDebugVCU118ShellPlacer(this, cJTAGDebugShellInput()))
   val jtagBScan = Overlay(JTAGDebugBScanOverlayKey, new JTAGDebugBScanVCU118ShellPlacer(this, JTAGDebugBScanShellInput()))
-  val fmc       = Overlay(PCIeOverlayKey, new PCIeVCU118FMCShellPlacer(this, PCIeShellInput()))
-  val edge      = Overlay(PCIeOverlayKey, new PCIeVCU118EdgeShellPlacer(this, PCIeShellInput()))
+  val fmc       = Overlay(PCIeFMCOverlayKey, new PCIeVCU118FMCShellPlacer(this, PCIeShellInput()))
```

Comment:
again, I don't understand the need for this

Reply:
For now, there is no any instanced PCIe IP in CoreIPFPGADesign.scala, so we get no PCIe interface for the current generated verilog RTL design.
I tried to add PCIe XDMA IP into design, but I always get FMC PCIe by the following description:
`val pcieinsts = p(PCIeEdgeOverlayKey).headOption.map(_.place(PCIeDesignInput(wranglerNode, corePLL = corePLL, bars = bars, ecam = ecam)).overlayOutput)`
Actually, what I need is to instance PCIe IP on Edge interface! So I separate PCIeOverlayKey to these two keys, I can use them to place IP by the project's needing.
But I think the PCIeOverlayKey is also can be kept if I can instance PCIe IP in the correct interface(FMC or Edge). Could you please also teach me how to do it? I just was bothered by this issue for 2 to 3 weeks, that's why I did it. Thank you again.

Reply:
I've reverted it to keep PCIeOverlayKey.